services/pull/merge_rebase.go

```diff
@@ -5,13 +5,101 @@ package pull
 
 import (
 	"fmt"
+	"strings"
 
 	repo_model "code.gitea.io/gitea/models/repo"
 	"code.gitea.io/gitea/modules/git"
 	"code.gitea.io/gitea/modules/log"
 )
 
-// doMergeStyleRebase rebaases the tracking branch on the base branch as the current HEAD with or with a merge commit to the original pr branch
+// Compose message to amend commit in rebase merge of pull request.
+func getRebaseAmendMessage(ctx *mergeContext, baseGitRepo *git.Repository) (message string, err error) {
+	// Get existing commit message.
+	var commitMessage strings.Builder
+	cmd := git.NewCommand(ctx, "show", "--format=%B", "-s")
+	if err := cmd.Run(&git.RunOpts{Dir: ctx.tmpBasePath, Stdout: &commitMessage}); err != nil {
+		return "", err
+	}
```

Comment:
A nit (not block),  git.NewCommand().RunStdString() could get the string output in a single line.

Reply:
Oh, I see the last commit. I meant `RunStdString` (not `Run`), then no need `commitMessage` or `Stdout: &commitMessage`

Reply:
Yes, I realized when I came to resolve the conversation here, fixed now.